src/AdminGuesser.js

```diff
@@ -16,51 +21,66 @@ const displayOverrideCode = resources => {
 };
 
 /**
- * AdminGuesserComponent automatically renders an `<Admin>` component for resources exposed by a web API documented with Hydra, OpenAPI or any other format supported by `@api-platform/api-doc-parser`.
+ * AdminGuesserComponent automatically renders an `<AdminUI>` component for resources exposed by a web API documented with Hydra, OpenAPI or any other format supported by `@api-platform/api-doc-parser`.
  * If child components are passed (usually `<ResourceGuesser>` or `<Resource>` components, but it can be any other React component), they are rendered in the given order.
- * If no children are passed, a `<ResourceGuesser>` component created for each resource type exposed by the API, in the order they are specified in the API documentation.
+ * If no children are passed, a `<ResourceGuesser>` component is created for each resource type exposed by the API, in the order they are specified in the API documentation.
  */
-export const AdminGuesserComponent = (
-  {children, includeDeprecated, ...props},
-  {data, error, loading},
-) => {
-  if (loading) {
-    return (
-      <TranslationProvider>
-        <Loading />
-      </TranslationProvider>
-    );
+export const AdminGuesserComponent = ({
+  children,
+  includeDeprecated,
+
+  resources,
+  fetching,
+  error,
+
+  ...rest
+}) => {
+  if (fetching) {
+    return <Loading />;
   }
 
-  if (error && data) {
+  if (error) {
     if ('production' === process.env.NODE_ENV) {
       console.error(error);
     }
-    return <div>Error while reading the API schema</div>;
+
+    return (
+      <ErrorUI
+        error={new Error('API schema is not readable')}
+        errorInfo={{componentStack: null}}
+      />
+    );
   }
 
-  if (!children && data) {
-    const resources = includeDeprecated
-      ? data.resources
-      : data.resources.filter(r => !r.deprecated);
-    children = resources.map(r => (
+  if (!children && resources) {
+    const guessResources = includeDeprecated
+      ? resources
+      : resources.filter(r => !r.deprecated);
+    children = guessResources.map(r => (
       <ResourceGuesser name={r.name} key={r.name} />
     ));
-    displayOverrideCode(resources);
+    displayOverrideCode(guessResources);
   }
 
-  return <Admin {...props}>{children}</Admin>;
+  return <AdminUI {...rest}>{children}</AdminUI>;
 };
 
-const AdminGuesser = props => (
-  <Query type={'INTROSPECT'}>
-    {state => AdminGuesserComponent(props, state)}
-  </Query>
-);
+const AdminGuesser = ({
+  dataProvider,
+  authProvider,
+  i18nProvider,
+  history = createHashHistory(),
+  customReducers,
+  customSagas,
+  initialState,
+
+  includeDeprecated = false,
 
-AdminGuesser.defaultProps = {
-  includeDeprecated: false,
-  theme: createMuiTheme({
+  appLayout,
+  layout = Layout,
+  loginPage,
+  locale,
+  theme = createMuiTheme({
```

Comment:
prefer `defaultProps` to default values in the component props signature, because `defaultProps` are only created once, whereas default values are created each time the component renders. You don't know how many times you component may render, and returning a new value for the `theme` prop will rerender the whole tree even if the two themes are shallow equal.

Reply:
It seems `defaultProps` will be deprecated for function component: https://github.com/reactjs/rfcs/pull/107.
That's why I chose to do it like this.
Maybe I can do the instanciation before though for performance reasons.

Reply:
yes, you'll need to move the createMuiTheme outside